accelerator/config.c

```diff
@@ -92,64 +91,63 @@ status_t cli_config_set(char* conf_file, ta_config_t* const info, iota_config_t*
     // File configuration
     case CONF_CLI: {
       size_t arg_len = strlen(value);
-      strncpy(conf_file, value, arg_len);
-      conf_file[arg_len] = '\0';
+      strncpy(core->conf_file, value, arg_len);
+      core->conf_file[arg_len] = '\0';
       break;
     }
   }
   return SC_OK;
 }
 
-status_t ta_config_default_init(ta_config_t* const info, iota_config_t* const iconf, ta_cache_t* const cache,
-                                iota_client_service_t* const service) {
+status_t ta_core_default_init(ta_core_t* core) {
   status_t ret = SC_OK;
 
   logger_id = logger_helper_enable(CONFIG_LOGGER, LOGGER_DEBUG, true);
   ta_log_info("enable logger %s.\n", CONFIG_LOGGER);
 
-  if (info == NULL || iconf == NULL || cache == NULL || service == NULL) {
-    ta_log_error("%s\n", "SC_TA_NULL");
+  if (core == NULL) {
+    ta_log_error("Invalid pointer(NULL) to ta_core_t\n");
     return SC_TA_NULL;
   }
 
   ta_log_info("Initializing TA information\n");
-  info->version = TA_VERSION;
-  info->host = TA_HOST;
-  info->port = TA_PORT;
-  info->thread_count = TA_THREAD_COUNT;
+  core->ta_conf.version = TA_VERSION;
```

Comment:
I am afraid that more references are required every time we attempt to access specific fields.

Reply:
That is true.
I used a single pointer to ta_core_t because of this kind of function defines all members in `ta_core_t` as their parameters. As I am going to introduce `db_service `to `ta_core_t `for ScyllaDB, I find all of those function calls need to be modified and become longer.

Reply:
Maybe we can use pointers to some members when we need to reduce references.
Or should I rollback this change?

Reply:
Let's minimize the changes.